src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1780,25 +1780,40 @@ private void SlideWindowToIncludeTab(int tabSelected)
             }
         }
 
-        private void Button_MouseEnter(object sender, MouseEventArgs e)
+        private void LibraryHandle_MouseEnter(object sender, MouseEventArgs e)
         {
             Grid g = (Grid)sender;
             StackPanel sp = (StackPanel)(g.Children[0]);
             TextBlock tb = (TextBlock)(sp.Children[0]);
-            var bc = new BrushConverter();
-            tb.Foreground = (Brush)bc.ConvertFrom("#cccccc");
             Image collapseIcon = (Image)sp.Children[1];
 
-            // When hovered swap appropriate expand/collapse button state
+            UpdateHandleHoveredStyle(tb, collapseIcon);
+        }
+
+        private void UpdateHandleHoveredStyle(TextBlock text, Image icon)
+        {
+            var bc = new BrushConverter();
+            text.Foreground = (Brush)bc.ConvertFrom("#cccccc");
+
             if (LibraryCollapsed || ExtensionsCollapsed)
             {
                 Uri imageUri;
                 imageUri = new Uri(@"pack://application:,,,/DynamoCoreWpf;component/UI/Images/expand_hover.png");
                 BitmapImage hover = new BitmapImage(imageUri);
-                collapseIcon.Source = hover;
+                icon.Source = hover;
             }
         }
 
+        private void ExtensionHandle_MouseEnter(object sender, MouseEventArgs e)
+        {
+            Grid g = (Grid)sender;
```

Comment:
Should we check the type of `sender` here `is Grid` before doing all of the following?

Reply:
This part did not really change. I would say it is fine like this given it is an event handler, and as such should not be reused.